### ISSUE TITLE
Storable.Mutable.set operation for Vector Double does not work correctly on ghcjs

### ISSUE THREAD

**Issue by dfordivam** [haskell/vector#410]
With ghcjs the below code results in output `[1.0,5.299808824e-315]`

```
import qualified Data.Vector.Storable as S
import qualified Data.Vector.Storable.Mutable as SM

vecSet = do
  v <- SM.new 2
  SM.set v (1.0 :: Double)
  print =<< S.freeze v
```

The problematic code is here, this causes the use of primitive `Word64` operations which dont work for `Double` as their underlying representation is different.

https://github.com/haskell/vector/blob/f2339ea0c588f6e3c11c563c1cb1e2dcfc316e16/vector/src/Data/Vector/Storable/Mutable.hs#L217

**Comment by lehins** [comment]
Fixed in #411

**Comment by lehins** [comment]
@dfordivam In case you need this fix, I just released a minor fixup version that includes this bugfix: [`vector-0.12.3.1`](https://hackage.haskell.org/package/vector-0.12.3.1)
